packages/connectivity/lib/connectivity.dart

```diff
@@ -12,7 +12,28 @@ import 'package:meta/meta.dart';
 /// WiFi: Device connected via Wi-Fi
 /// Mobile: Device connected to cellular network
 /// None: Device not connected to any network
-enum ConnectivityResult { wifi, mobile, none }
+enum ConnectionType { wifi, mobile, none }
+enum ConnectionSubtype { EDGE, HSDPA, LTE, none, unknown }
```

Comment:
maybe fully expose all the subtypes from the platform instead of translate into only these 3.